firebase_remote_config/lib/src/sentry_firebase_remote_config_integration.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:async';
+
+import 'package:firebase_remote_config/firebase_remote_config.dart';
+import 'package:sentry/sentry.dart';
+
+class SentryFirebaseRemoteConfigIntegration extends Integration<SentryOptions> {
+  SentryFirebaseRemoteConfigIntegration({
+    required FirebaseRemoteConfig firebaseRemoteConfig,
+    bool activateOnConfigUpdated = true,
+  })  : _firebaseRemoteConfig = firebaseRemoteConfig,
+        _activateOnConfigUpdated = activateOnConfigUpdated;
+
+  final FirebaseRemoteConfig _firebaseRemoteConfig;
+  final bool _activateOnConfigUpdated;
+  StreamSubscription<RemoteConfigUpdate>? _subscription;
+
+  @override
+  FutureOr<void> call(Hub hub, SentryOptions options) async {
+    _subscription = _firebaseRemoteConfig.onConfigUpdated.listen((event) async {
+      if (_activateOnConfigUpdated) {
+        await _firebaseRemoteConfig.activate();
+      }
+      for (final updatedKey in event.updatedKeys) {
+        final value = _firebaseRemoteConfig.getBoolOrNull(updatedKey);
+        if (value != null) {
+          await Sentry.addFeatureFlag(updatedKey, value);
+        }
+      }
+    });
+    options.sdk.addIntegration('SentryFirebaseRemoteConfigIntegration');
+  }
+
+  @override
+  FutureOr<void> close() async {
+    await _subscription?.cancel();
+    _subscription = null;
+  }
+}
+
+extension _SentryFirebaseRemoteConfig on FirebaseRemoteConfig {
+  bool? getBoolOrNull(String key) {
+    final strValue = getString(key);
+    final lowerCase = strValue.toLowerCase();
+    if (lowerCase == 'true' || lowerCase == '1') {
+      return true;
+    }
+    if (lowerCase == 'false' || lowerCase == '0') {
+      return false;
+    }
```

Comment:
I don't think we should check for the numerical value, 1 or 0 could also represent something else in the user's feature flag and not necessarily a boolean

If I'm not wrong then a 'true' or 'false' string are the only times we really know that those are bools (aside from the user actually setting the string to those values) right?

Reply:
This is the same check as in the [dependency](https://pub.dev/documentation/firebase_remote_config/latest/firebase_remote_config/RemoteConfigValue/asBool.html) for bool values. I'm not sure how firebase really saves the value. So we could miss a boolean value that's encoded as `0` or `1`.

<img width="685" alt="Bildschirmfoto 2025-04-09 um 16 01 05" src="https://github.com/user-attachments/assets/1b4b5957-d997-4e7c-9cc2-85fe5b230f4f" />

Reply:
And judging from the [default value](https://pub.dev/documentation/firebase_remote_config/latest/firebase_remote_config/RemoteConfigValue/defaultValueForDouble-constant.html), it could rather be 0/1 than `true/false`.

Reply:
You could try out checking what the strValue is for a bool, I don't think the bool representation is going to be inconsistent

Reply:
alright then let's leave it as it is, let's keep this in mind in case this ever comes up